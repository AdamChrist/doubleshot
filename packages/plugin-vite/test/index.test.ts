import path from 'path'
import { afterAll, beforeAll, describe, expect, it } from 'vitest'
import { execa } from 'execa'
import fs from 'fs-extra'
const mockDir = path.resolve(__dirname, './mock')

const installDeps = async (cwd: string) => {
  const { stdout, stderr } = await execa(
    'npm',
    ['install', '--package-lock=false'],
    {
      cwd,
    },
  )

  const logs = stdout + stderr
  return logs
}

const remove = () => {
  fs.removeSync(path.resolve(mockDir, 'dist'))
}

const run = async (mode: 'dev' | 'build') => {
  const { stdout, stderr } = await execa(
    'npm',
    ['run', mode],
    {
      cwd: mockDir,
    },
  )

  const logs = stdout + stderr
  return logs
}

beforeAll(async () => {
  remove()
  await installDeps(mockDir)
}, Infinity)

afterAll(() => {
  remove()
})

describe('Doubleshot Vite Plugin', () => {
  it('should run dev mode', async () => {
    const logs = await run('dev')

    expect(logs).toContain('Development')
  })

  it('should also inject preload', async () => {
    const logs = await run('dev')

    expect(logs).toContain('hello, this is a msg through preload file')
  })

  it('should build electron app if "electron.build" is set', async () => {
    const logs = await run('build')

    expect(logs).toContain('Start electron build')
    expect(logs).toContain('Build succeeded')
    expect(fs.existsSync(path.resolve(mockDir, 'dist/electron'))).toBe(true)
  }, 10 * 60 * 1000)
})
