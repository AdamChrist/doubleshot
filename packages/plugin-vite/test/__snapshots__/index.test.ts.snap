// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Doubleshot Vite Plugin > plugin's config should be overridden by \`configureForMode\` > should match snapshot in development mode 1`] = `
"\\"use strict\\";

// src/index.ts
var import_path = require(\\"path\\");
var import_electron = require(\\"electron\\");
var createWindow = () => {
  const isDev = !import_electron.app.isPackaged;
  const win = new import_electron.BrowserWindow({
    width: 800,
    height: 600,
    webPreferences: {
      contextIsolation: true,
      preload: (0, import_path.join)(__dirname, \\"./preload.js\\")
    }
  });
  import_electron.ipcMain.handle(\\"msg\\", (_, msg) => {
    console.log(msg);
    import_electron.app.quit();
  });
  const URL = isDev ? \\"http://127.0.0.1:5173\\" : \`file://\${(0, import_path.join)(import_electron.app.getAppPath(), \\"dist/renderer/index.html\\")}\`;
  win.loadURL(URL);
};
import_electron.app.whenReady().then(() => {
  createWindow();
});
import_electron.app.on(\\"window-all-closed\\", () => {
  if (process.platform !== \\"darwin\\")
    import_electron.app.quit();
});
"
`;

exports[`Doubleshot Vite Plugin > plugin's config should be overridden by \`configureForMode\` > should match snapshot in production mode 1`] = `
"\\"use strict\\";var import_path=require(\\"path\\");var import_electron=require(\\"electron\\");var createWindow=()=>{const isDev=!import_electron.app.isPackaged;const win=new import_electron.BrowserWindow({width:800,height:600,webPreferences:{contextIsolation:true,preload:(0,import_path.join)(__dirname,\\"./preload.js\\")}});import_electron.ipcMain.handle(\\"msg\\",(_,msg)=>{console.log(msg);import_electron.app.quit()});const URL=isDev?process.env.DS_RENDERER_URL:\`file://\${(0,import_path.join)(import_electron.app.getAppPath(),\\"dist/renderer/index.html\\")}\`;win.loadURL(URL)};import_electron.app.whenReady().then(()=>{createWindow()});import_electron.app.on(\\"window-all-closed\\",()=>{if(process.platform!==\\"darwin\\")import_electron.app.quit()});
"
`;

exports[`Doubleshot Vite Plugin > should also inject preload 1`] = `
"\\"use strict\\";var import_path=require(\\"path\\");var import_electron=require(\\"electron\\");var createWindow=()=>{const isDev=!import_electron.app.isPackaged;const win=new import_electron.BrowserWindow({width:800,height:600,webPreferences:{contextIsolation:true,preload:(0,import_path.join)(__dirname,\\"./preload.js\\")}});import_electron.ipcMain.handle(\\"msg\\",(_,msg)=>{console.log(msg);import_electron.app.quit()});const URL=isDev?process.env.DS_RENDERER_URL:\`file://\${(0,import_path.join)(import_electron.app.getAppPath(),\\"dist/renderer/index.html\\")}\`;win.loadURL(URL)};import_electron.app.whenReady().then(()=>{createWindow()});import_electron.app.on(\\"window-all-closed\\",()=>{if(process.platform!==\\"darwin\\")import_electron.app.quit()});
"
`;
